vesc_driver/launch/vesc_driver_node.launch.py

```diff
@@ -37,22 +37,43 @@
 
 def generate_launch_description():
 
-    vesc_config = os.path.join(
+    # Get default config file.
+    config = os.path.join(
         get_package_share_directory('vesc_driver'),
         'params',
-        'vesc_config.yaml'
+        'config.yaml'
         )
-    return LaunchDescription([
-        DeclareLaunchArgument(
+
+    # Create the launch configuration variables
+    config_yaml = LaunchConfiguration("config")
+    log_level = LaunchConfiguration('log_level')
```

Comment:
Please move these below the associated `DeclareLaunchArgument` lines below so they are not used before they are delcared.

Reply:
I am not an expert of the launch file in ros2 but I assume that the real declaration of the arguments is done by the caller of the `generate_launch_description` when processing the returned `LaunchDescription` ... so I am not sure that the order is actually important here.

What I did was try to follow what is done in the nav2 package ( https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/launch/bringup_launch.py ) ... they are far from perfect but probably the code should already have quite some user that are testing it